### PR TITLE
Support multiple paths per spec

### PR DESCRIPTION
**Pull request by ritvik-statsig**

### CODE REVIEW

**Comment by ritvik-statsig** [comment]
@shawncao - in response to a few of your comments, the grouping on paths isn't necessarily going to be stable. it's possible for it to change because listing paths isn't guaranteed to return in the same order. Will that be a problem?

**Comment by shawncao** [comment]
> 

what about guaranteeing the list files to be stable for any given folder? for example, `list` method will always return a sorted order?

File: src/ingest/IngestSpec.cpp [hunk 1]
```diff
@@ -489,25 +508,31 @@ bool IngestSpec::ingest(const std::string& file, BlockList& blocks) noexcept {
...
+      if (table_->format == DataFormat::CSV) {
+        source = std::make_unique<CsvReader>(file, table_->csv, columns);
+      } else if (table_->format == DataFormat::JSON) {
+        source = makeJsonReader(file, table_->json, schema);
+      } else if (table_->format == DataFormat::PARQUET) {
+        // schema is modified with time column, we need original schema here
+        source = std::make_unique<ParquetReader>(file, schema);
+      } else {
+        LOG(ERROR) << "Supported data formats: csv, json, parquet.";
+        continue;
```
[the remark this answers is not in the document]
Comment (shawncao) [reply src/ingest/IngestSpec.cpp:534]: https://github.com/varchar-io/nebula/issues/175

File: src/ingest/IngestSpec.h [hunk 2]
```diff
@@ -61,26 +61,26 @@ class IngestSpec : public nebula::common::Identifiable {
...
     size_t size,
     SpecState state,
     size_t watermark,
     nebula::common::unordered_map<std::string, std::string> macroCombinations = nebula::common::unordered_map<std::string, std::string>())
     : table_{ table },
       version_{ version },
-      path_{ path },
+      paths_{ paths },
       domain_{ domain },
       size_{ size },
```
[the remark this answers is not in the document]
Comment (shawncao) [reply src/ingest/IngestSpec.h:74]: https://github.com/varchar-io/nebula/issues/176

**Pull request merged**